parsl/monitoring/db_manager.py

```diff
@@ -539,7 +544,20 @@ def _migrate_logs_to_internal(self, logs_queue: queue.Queue, queue_tag: str, kil
 
     def _update(self, table: str, columns: List[str], messages: List[Dict[str, Any]]) -> None:
         try:
-            self.db.update(table=table, columns=columns, messages=messages)
+            done = False
+            while not done:
+                try:
+                    self.db.update(table=table, columns=columns, messages=messages)
+                    done = True
+                except sa.exc.OperationalError as e:
+                    # This code assumes that an OperationalError is something that will go away eventually
```

Comment:
This assumption may be really bad? Probably set a maximum retry like 5 times?

Reply:
Maximum retry counts don't solve the problem this PR is intended to address: if I have a big query running for 10 minutes, I want parsl to buffer the data rather than discard it.

Perhaps it is fine to keep trying through the duration of the workflow run, perhaps with some backoff to once a minute or so, with better handling of shutdown - the code at the moment won't shut down if it gets stuck in an operationalerror loop.